x/dex/types/pool_reserves.go

```diff
@@ -5,5 +5,37 @@ import (
 )
 
 func (p PoolReserves) HasToken() bool {
-	return p.Reserves.GT(sdk.ZeroInt())
+	return p.ReservesMakerDenom.GT(sdk.ZeroInt())
+}
+
+func NewPoolReserves(makerDenom string, takerDenom string, tickIndex int64, fee uint64, reservesMakerDenom sdk.Int) (*PoolReserves, error) {
```

Comment:
Nit: if we're only using this for tests, let's keep it in the test library. 
Also, I don't love the duplication of code between this and GetOrInItPoolReserves.